app/map/currentlocation.js

```diff
@@ -0,0 +1,62 @@
+// Note: User must consent to location sharing when prompted by their browser. 
+// If they see the error "The Geolocation service failed.", 
+// it means that they most probably did not give permission for the browser to access their current location.
+
+let map, infoWindow;
+
+function initMap() {
+  map = new google.maps.Map(document.getElementById("map"), {
+    center: { lat: 1, lng: 1},
+    zoom: 6,
+  });
+  infoWindow = new google.maps.InfoWindow();
+
+  const locationButton = document.createElement("button");
+
+  locationButton.textContent = "Pan to Current Location";
+  locationButton.classList.add("custom-map-control-button");
+  map.controls[google.maps.ControlPosition.TOP_CENTER].push(locationButton);
+  locationButton.addEventListener("click", () => {
+    // Try HTML5 geolocation.
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition(
+        (position) => {
+          const pos = {
+            lat: position.coords.latitude,
+            lng: position.coords.longitude,
+          };
+
+          infoWindow.setPosition(pos);
+          infoWindow.setContent("Location found.");
+          infoWindow.open(map);
+          map.setCenter(pos);
+        },
+        () => {
+          handleLocationError(true, infoWindow, map.getCenter());
+        }
+      );
+    } else {
+      // Browser doesn't support Geolocation
+      handleLocationError(false, infoWindow, map.getCenter());
+    }
+  });
+}
+
+//must set specific locations for all our racks in lat and lng
+const myLatLng = { lat: 1, lng: 1};
+new google.maps.Marker({
```

Comment:
that being said, from the db, you would need lat and lng info associated with the stations?

Reply:
Parameters like `location` accept latitude and longitude as comma-separated values (x,y), so then we can set that as the name in the database and load things in that way. So for the lat and lng, it would just by (lat, lng).